realm/src/main/java/io/realm/internal/Group.java

```diff
@@ -217,7 +217,8 @@ public Table getTable(String name) {
         }
         if (immutable) {
             if (!hasTable(name)) {
-                throwImmutable();
+                throw new IllegalStateException("Requested table is not in this Realm. " +
```

Comment:
I think that the table name (`name`) should be place earlier. In the lines of: "Requested table 'name' is not in this Realm. Creating the table must be done within a transaction."